### ISSUE TITLE
ConnectionResetError: [WinError 10054] when robot or viewer call socket recv.

### ISSUE THREAD

**Issue by dbakewel** [dbakewel/netbots#2]
**Describe the bug**
Some of the processes stated by rundemo.bat crash with error on Ms. Wear's laptop. The problem persists after a reboot.

**To Reproduce**
Double click rundemo.exe on Ms. Wear's laptop. The issue has not been seen elsewhere.

**Expected behavior**
The demo should start normally.

**Console Output**
```
INFO 2019-04-09 07:22:11.946 netbots_log.setLogLevel(): DEBUG logging = False. VERBOSE logging = False
Traceback (most recent call last):
  File "robots/wallbanger.py", line 118, in <module>
    main()
  File "robots/wallbanger.py", line 100, in main
    joinReply = botSocket.sendRecvMessage({'type': 'joinRequest', 'name': robotName})
  File "C:\Users\Lara\Documents\teaching\APCS\netbots-mastern.new\netbots-master\src\netbots_ipc.py", line 365, in sendRecvMessage
    replyMsg, ip, port = self.recvMessage()
  File "C:\Users\Lara\Documents\teaching\APCS\netbots-mastern.new\netbots-master\src\netbots_ipc.py", line 286, in recvMessage
    bytesAddressPair = self.s.recvfrom(self.bufferSize)
ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host
```
**Work Around**
Changing the port numbers in rundemo.bat fixes the issue.

**Comment by dbakewel** [comment]
The error message is misleading because UDP protocol doesn't not have connections. What is happening is that the UDP socket on the receiving end is not open but the receiving OS is running on the destination IP. In this case the robot sends a message to the netbots server before the server has opened its port. The receiving OS sends back an ICMP unreachable message since the port is not open. On Linux, the python code is not aware of this as no exception is raised. On Windows however, a ConnectionResetError exception is raised. This should be caught by the netbots_ipc module and ignored.

See example tcpdump output from server below. Server at 192.168.1.1 but not running, robot sending join message from 192.168.1.2.
```
10:02:49.406979 IP 192.168.1.2.20010 > 192.168.1.1.20000: UDP, length 32
10:02:49.407040 IP 192.168.1.1 > 192.168.1.2: ICMP 192.168.8.241 udp port 20000 unreachable, length 68
```

**Comment by dbakewel** [comment]
This is fixed in commit b3619b8378bcab277110ac2846ae43a1ef3b359c
